### PR TITLE
Updates for MkDocs v1

### PR DESCRIPTION
**Pull request by matyasselmeci**
(SOFTWARE-4141)

### CODE REVIEW

File: docs/documentation/style-guide.md [hunk 1]
```diff
@@ -28,16 +28,24 @@ Use the following conventions for headings:
...
+    MkDocs 1.0.0 does not support site-relative links (e.g. `/software/development-process`).
+    You must use document-relative links (e.g. `../software/development-process`) instead.
+
+    Earlier versions of this guide recommended site-relative links;
+    these only worked in earlier versions of MkDocs.
+
+    Please convert any site-relative links to document-relative links
+    before updating the `doc-ci-scripts` submodule to use MkDocs 1.0.0+.
 
 - Links to internal pages should not have the `.md` extension
```
Comment (brianhlin) [inline docs/documentation/style-guide.md:41]:
```suggestion
```

The MkDocs documentation uses the `.md` suffix so let's go with that. Though I think it's fine either way

File: docs/documentation/writing-documentation.md [hunk 2]
```diff
@@ -148,6 +155,58 @@ $ git push upstream new_docs:itb.new_docs
...
+However, we are planning to upgrade each repo to use MkDocs 1.0.4 or newer,
+which contains additional functionality and bug fixes.
+
+Before upgrading, you must fix the following incompatibilities:
+
+- Change the `pages:` section of `mkdocs.yml` to a `nav:` section.
+  The section contents are identical; only the name is changing.
+
+- Update all of the links in the documents as follows:
+  - Ensure links do not end in `.md`
```
Comment (brianhlin) [inline docs/documentation/writing-documentation.md:173]:
```suggestion
```

File: docs/documentation/writing-documentation.md [hunk 3]
```diff
@@ -148,6 +155,58 @@ $ git push upstream new_docs:itb.new_docs
...
+  - Ensure links are document-relative, not site-relative (e.g. `../software/development-process`
+    instead of `/software/development-process`
+
+Once these are done, use the following procedure to update:
+
+1. Create an itb branch from master, e.g.
```
Comment (brianhlin) [inline docs/documentation/writing-documentation.md:179]: We should have a step or sentence here indicating that this needs to be done from within the repo they're updating.

File: docs/documentation/style-guide.md (hunk 1 above)
Comment (matyasselmeci) [reply docs/documentation/style-guide.md:41]: The JIRA ticket says the opposite:
> Any internal links should not have the .md suffix

Comment (brianhlin) [reply docs/documentation/style-guide.md:41]: Argh, sorry. That was an oversight on my part. Further down in the description I added:

> Fix internal technology links (they need to be doc-relative with .md suffixes according to https://www.mkdocs.org/user-guide/writing-your-docs/#linking-to-pages

Comment (matyasselmeci) [reply docs/documentation/style-guide.md:41]: Wait, do we _need_ to add the .md suffixes everywhere? Turning `](/` into `](../` was pretty easy, but the other one is going to be a pretty grody regex to write...

Comment (brianhlin) [reply docs/documentation/style-guide.md:41]: I *think* it works without the suffix so I'm open to excluding that requirement

Comment (matyasselmeci) [reply docs/documentation/style-guide.md]: Tested it and no, it really doesn't. I modified the text to add the md requirement.

**Review by brianhlin — APPROVED** [review]
LGTM!

**Pull request merged**